tests/test_local_sandbox.py

```
@@ -0,0 +1,240 @@
+    assert isinstance(sandbox, Sandbox)
+    typed: Sandbox = sandbox  # static conformance, checked because tests are type-checked
+    assert typed.provider == 'local'
+    assert sandbox.sandbox_id.startswith('local-')
+
+
+async def test_run_argv_and_shell(tmp_path: Path):
+    sandbox = LocalSandbox(tmp_path)
+    result = await sandbox.run(['echo', 'hello'])
+    assert (result.exit_code, result.stdout, result.stderr) == (0, 'hello\n', '')
+    assert result.stdout_dropped == 0 and result.stderr_dropped == 0
+    shell_result = await sandbox.run('echo foo | tr a-z A-Z', shell=True)
```

Ruff (0.15.21)

[error] Function call with `shell=True` parameter identified, security issue

(S604)